pkg/tcpip/transport/tcp/accept.go

```diff
@@ -515,7 +518,10 @@ func (e *endpoint) handleListenSegment(ctx *listenContext, s *segment) {
 		n.tsOffset = 0
 
 		// Switch state to connected.
-		n.state = StateEstablished
+		if n.state != StateEstablished {
+			n.stack.Stats().TCP.CurrentEstablished.Increment()
+			n.state = StateEstablished
+		}
 
```

Comment:
The check is redundant. At this point the connection cannot have been in an ESTABLISHED state.

Reply:
OK, I'll remove the check. Thanks!

Reply:
These checks are still present?